homeassistant/components/vicare/number.py

```diff
@@ -143,9 +193,24 @@ def update(self) -> None:
         """Update state of number."""
         try:
             with suppress(PyViCareNotSupportedFeatureError):
-                self._attr_native_value = self.entity_description.value_getter(
-                    self._api
+                self._attr_native_value = float(
```

Comment:
Why the addition of float? This should return a number already or None (which would not work)?

Reply:
[The getter](https://github.com/CFenner/home-assistant-core/blob/46322a0f5988f2ee68918113e798bebd5e236993/homeassistant/components/vicare/__init__.py#L39) is defined in the interface to return a bool.
But the lambda returns a float, that's true.

Reply:
Then I suppose you should overwrite it here or probably fix the base entity description as it can clearly return not only bool.

Reply:
Actually the library is not really types and the function just return `Any`. Is it now better to use `Any` as well or assume the types based on the api description?
Probably on the long run the library should be typed.